pkg/sql/sem/builtins/generator_builtins.go

```diff
@@ -1231,6 +1241,141 @@ func (g *jsonEachGenerator) Values() (tree.Datums, error) {
 	return tree.Datums{g.key, g.value}, nil
 }
 
+var jsonPopulateProps = tree.FunctionProperties{
+	Class:    tree.GeneratorClass,
+	Category: categoryGenerator,
+	// The typical way to call json_populate_record is to send NULL::atype as the
+	// first argument, so we have to accept nullable args.
+	NullableArgs: true,
+}
+
+func makeJSONPopulateImpl(gen tree.GeneratorFactory, info string) tree.Overload {
+	return tree.Overload{
+		Types: tree.ArgTypes{{"base", types.Any}, {"from_json", types.Jsonb}},
+		ReturnType: func(args []tree.TypedExpr) *types.T {
+			if len(args) != 2 {
+				return tree.UnknownReturnType
+			}
+			return args[0].ResolvedType()
+		},
+		Generator: gen,
+		EvalArgs: func(ctx *tree.EvalContext, exprs tree.Exprs) (tree.Datums, error) {
```

Comment:
Can you provide some better commentary on the structure of the args here? It's not trivial

Reply:
Yes, I'll add some more coments.